dcs/core/data_repository.py

```diff
@@ -212,3 +215,44 @@ async def serve_envelope(self, *, drs_id: str, public_key: str) -> str:
             raise self.EnvelopeNotFoundError(object_id=drs_id) from error
 
         return envelope
+
+    async def delete_file(self, *, file_id: str) -> None:
+        """Deletes a file from the outbox storage, the internal database and the
+        corresponding secret from the secrets store.
+        If no file or secret with that id exists, do nothing.
+
+        Args:
+            file_id: id for the file to delete.
+        """
+
+        # get secret id
+        drs_object = await self._drs_object_dao.get_by_id(id_=file_id)
```

Comment:
This should be in a try/except ResourceNotFoundError, else it might crash the service on bogus/already deleted IDs